### PR TITLE
Adding support for GET /apa/api-findings/v1/finding endpoint

### PR DESCRIPTION
**Pull request by naumraviz**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Adding support for GET /apa/api-findings/v1/finding endpoint**

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Python Version(s) Tested: 3.11.6
* Tenable.sc version (if necessary):

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tenable/apa/findings/api.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    def _get_page(self) -> None:
+        '''
+        Request the next page of data
+        '''
+        payload = copy(self._payload)
+        print(self._next_token)
+        payload['next'] = self._next_token
+
+        if self.num_pages >= 1 and self._next_token is None:
+            raise StopIteration()
```
[the remark this answers is not in the document]
Comment (naumraviz) [reply tenable/apa/findings/api.py:37]: I removed that condition 
I don't think I need it (I copied it from somewhere else)

I want to clarify 
Does the iterator stop when the received findings count equals the total findings?

File: tenable/apa/session.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        >>> tapa = TenableAPA('ACCESS_KEY', 'SECRET_KEY')
+
+        Example with proper identification:
+
+        >>> tapa = TenableAPA('ACCESS_KEY', 'SECRET_KEY',
+        >>>     vendor='Company Name',
+        >>>     product='My Awesome Widget',
+        >>>     build='1.0.0')
+    '''
+
```
[the remark this answers is not in the document]
Comment (naumraviz) [reply tenable/apa/session.py:29]: Done

File: tests/apa/findings/test_findings_api.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+def test_findings_list_findings_page_response(api, findings_page_response):
+
+    responses.get('https://cloud.tenable.com/apa/findings-api/v1/findings',
+                  json=findings_page_response)
+
+    findings_page: FindingsPageSchema = api.findings.list(return_iterator=False)
+
+    assert findings_page == FindingsPageSchema().load(findings_page_response)
+    assert len(responses.calls) == 1
+    assert responses.calls[0].request.url == "https://cloud.tenable.com/apa/findings-api/v1/findings?limit=50"
```
[the remark this answers is not in the document]
Comment (naumraviz) [reply tests/apa/findings/test_findings_api.py:180]: Done

File: tests/apa/findings/test_findings_api.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+    for f in findings:
+        assert f == findings_page_response['data'][0]
+
+    assert len(responses.calls) == 28
+    for i, c in enumerate(responses.calls):
+        if i == 0:
+            assert c.request.url == 'https://cloud.tenable.com/apa/findings-api/v1/findings?limit=1&filter=value&sort_filed=last_updated_at&sort_order=desc'
+        else:
+            assert c.request.url == 'https://cloud.tenable.com/apa/findings-api/v1/findings?next=eyJsYXN0X3NlcmlhbF9pZCI6IDE0MTE2NTEsICJvcmRlcl9ieV9maWVsZF9uYW1lIjogbnVsbCwgImxhc3Rfb3JkZXJfYnlfdmFsdWUiOiBudWxsfQ%3D%3D&limit=1&filter=value&sort_filed=last_updated_at&sort_order=desc'
```
[the remark this answers is not in the document]
Comment (naumraviz) [reply tests/apa/findings/test_findings_api.py:167]: Done

**Comment by naumraviz** [comment]
I fixed flake8, but it still says, when I ran black, it create flake erros
image
image
image

File: tenable/apa/findings/api.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (naumraviz) [reply tenable/apa/findings/api.py]: I do return the next token on the last page - I improved the tests

**Review by SteveMcGrath — APPROVED** [review]

**Pull request merged**